Userland/Applications/TodoList/TaskListModel.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, Arthur Brainville <ybalrid@ybalrid.info>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "TaskListModel.h"
+
+#include <AK/JsonArray.h>
+#include <AK/JsonObject.h>
+#include <AK/JsonParser.h>
+#include <AK/JsonValue.h>
+
+#include <cstdio>
+
+void TaskListModel::load_from_disk(AK::String path)
+{
+    //TODO There's probably a better API in Serenity than the stdio for that
+    FILE* todofile = fopen(path.characters(), "r");
+    StringBuilder jsonBuilder;
+    if (todofile) {
+        for (;;) {
+            char buffer[1024];
+            if (!fgets(buffer, sizeof(buffer), todofile))
+                break;
+            jsonBuilder.append(buffer);
+        }
+
+        AK::JsonValue todoJson = AK::JsonValue::from_string(jsonBuilder.to_string()).value();
+
+        if (todoJson.is_array()) {
+            auto list = todoJson.as_array();
+            for (size_t i = 0; i < list.size(); ++i) {
+                const auto& task = list[i].as_object();
```

Comment:
The nesting level here is pretty high. Maybe return early in case of an error while opening the file or when the file isn't valid JSON.

Reply:
Added an early returns, now this has been flattened to nested to only one level in the for loop